src/Fixer/Comment/HeaderCommentFixer.php

```diff
@@ -126,43 +129,81 @@ private function findHeaderCommentIndex(Tokens $tokens)
     private function findHeaderCommentInsertionIndex(Tokens $tokens)
     {
         $index = $tokens->getNextNonWhitespace(0);
+        if (null !== $index && $tokens[$index]->isGivenKind(T_DECLARE)) {
+            $index = $this->skipDeclare($tokens, $index);
+        }
 
         if (null === $index) {
             // empty file, insert at the end
             $index = $tokens->getSize();
         }
 
-        return $index;
+        return $index === 2 && $tokens[1]->isWhitespace() ? 1 : $index;
+    }
+
+    /**
+     * Skips a declare(strict_type=1); statement as it is allowed to be before the header.
+     *
+     * @param Tokens $tokens
+     * @param int    $index
+     *
+     * @return int
+     */
+    private function skipDeclare(Tokens $tokens, $index)
+    {
+        $isOpen = false;
+        $isStrictTypes = false;
+        $originalIndex = $index;
+
+        while ($index = $tokens->getNextNonWhitespace($index)) {
+            if ($tokens[$index]->getContent() === '(') {
+                $isOpen = true;
+            }
+            if ($isOpen && $tokens[$index]->getContent() === 'strict_types') {
+                $isStrictTypes = true;
+            }
+            if ($tokens[$index]->getContent() === ';') {
+                ++$index;
+                break;
+            }
+        }
+
+        return $isStrictTypes ? $index : $originalIndex;
```

Comment:
@keradus but you example will not pass the check, because there is a `;` before `strict_types`.. ;)